src/hotspot/share/gc/shenandoah/shenandoahGenerationalHeap.cpp

```diff
@@ -216,7 +216,7 @@ oop ShenandoahGenerationalHeap::evacuate_object(oop p, Thread* thread) {
     if (mark.has_displaced_mark_helper()) {
       // We don't want to deal with MT here just to ensure we read the right mark word.
       // Skip the potential promotion attempt for this one.
-    } else if (r->age() + mark.age() >= age_census()->tenuring_threshold()) {
+    } else if (age_census()->is_tenurable(r->age() + mark.age())) {
```

Comment:
See comment above re using this boolean accessor at other places where comparisons are made with tenuring threshold.